bench/FileConnectionBench.java

```diff
@@ -0,0 +1,48 @@
+package benchmark;
+
+import com.sun.cldchi.jvm.JVM;
+import com.sun.midp.crypto.SecureRandom;
+import java.io.OutputStream;
+import java.util.Random;
+import javax.microedition.io.Connector;
+import javax.microedition.io.file.FileConnection;
+import org.mozilla.MemorySampler;
+
+public class FileConnectionBench {
+  void runBenchmark() {
+    try {
+      long start, time;
+
+      start = JVM.monotonicTimeMillis();
+
+      String privateDir = System.getProperty("fileconn.dir.private");
+      Random random = new Random();
+      String filename = new javax.bluetooth.UUID(java.lang.Math.abs(random.nextInt())).toString();
+      FileConnection file = (FileConnection)Connector.open(privateDir + filename);
+      System.out.println("Writing to file " + privateDir + filename);
+      file.create();
+
+      SecureRandom rnd = SecureRandom.getInstance(SecureRandom.ALG_SECURE_RANDOM);
+      OutputStream out = file.openOutputStream();
+      byte[] b = new byte[1024];
+      for (int i = 0; i < 1000; i++) {
+        rnd.nextBytes(b, 0, 1024);
```

Comment:
This is more of a memory benchmark. Nevertheless, I do report elapsed time, so it makes sense to focus that measurement on the subject of the benchmark. (And in any case I want to exclude the memory used by the random data generator from the memory benchmark.) In 5979324 I make that change.